### PR TITLE
align django requirement specs & allow 1.9.x patch releases, remove u…

### PR DESCRIPTION
**Pull request by DanCech**
…nused tox envs

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -35,7 +35,7 @@
 #   deactivate
 #
 
-Django>=1.9,<1.9.99
+Django>=1.9,<1.10
```
Comment (DanCech) [inline requirements.txt]: that is interesting, and somewhat weird.  I'll update this patch to align them all to be that way

**Comment by DanCech** [comment]
hmm, looks like travis-ci must have a hardcoded list of tox envs to run, since it's not happy about me nuking the unused ones.

**Comment by obfuscurity** [comment]
Yeah you need to update [.travis.yml](https://github.com/graphite-project/graphite-web/blob/master/.travis.yml) too.

**Comment by obfuscurity** [comment]
@cbowman0 @SEJeff @brutasse Any concerns with removing these?

**Comment by cbowman0** [comment]
I was thinking we should, so no concerns from me.

File: requirements.txt (hunk 1 above)
Comment (SEJeff) [inline requirements.txt]: I think the best way would be using the [compatible release operator](https://www.python.org/dev/peps/pep-0440/#compatible-release) here actually. Strangely no one on that mailing list thread realized you can use it:

    Django~=1.9.0

Is equivalent to asking for Django 1.9.*

**Comment by SEJeff** [comment]
@obfuscurity: my preference is to use `Django~=1.9.0` as it is a bit more elegant. As to removing old unsupported and security problematic versions of Django, strong :+1:

**Comment by DanCech** [comment]
@SEJeff thanks for the tip, I'll update the PR

**Comment by obfuscurity** [comment]
@SEJeff Do you have a link to that versioning feature?

**Comment by SEJeff** [comment]
Yessir! It is in [PEP-0440](https://www.python.org/dev/peps/pep-0440/#compatible-release)

**Comment by SEJeff** [comment]
`~=1.9` matches `1.*` whereas `~=1.9.0` matches `1.9.*`. That is how the operator works. The pep was cowritten by the pip author as well.

**Comment by obfuscurity** [comment]
Python, thank you for my fevered dreams tonight. 🤒

**Comment by obfuscurity** [comment]
This breaks pip requirements for older (e.g. system pip on 14.04).

```
vagrant@vagrant:~$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 14.04.2 LTS
Release:	14.04
Codename:	trusty

vagrant@vagrant:~$ pip -V
pip 1.5.4 from /usr/lib/python2.7/dist-packages (python 2.7)
```
```
+ cd ../graphite-web
+ git checkout master
Already on 'master'
Your branch is up-to-date with 'origin/master'.
+ pip install -r requirements.txt
Exception:
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/pip/basecommand.py", line 122, in main
    status = self.run(options, args)
  File "/usr/lib/python2.7/dist-packages/pip/commands/install.py", line 262, in run
    for req in parse_requirements(filename, finder=finder, options=options, session=session):
  File "/usr/lib/python2.7/dist-packages/pip/req.py", line 1632, in parse_requirements
    req = InstallRequirement.from_line(line, comes_from, prereleases=getattr(options, "pre", None))
  File "/usr/lib/python2.7/dist-packages/pip/req.py", line 173, in from_line
    return cls(req, comes_from, url=url, prereleases=prereleases)
  File "/usr/lib/python2.7/dist-packages/pip/req.py", line 71, in __init__
    req = pkg_resources.Requirement.parse(req)
  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 2667, in parse
    reqs = list(parse_requirements(s))
  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 2605, in parse_requirements
    line, p, specs = scan_list(VERSION,LINE_END,line,p,(1,2),"version spec")
  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 2573, in scan_list
    raise ValueError("Expected "+item_name+" in",line,"at",line[p:])
ValueError: ('Expected version spec in', 'Django~=1.9.0', 'at', '~=1.9.0')

Storing debug log for failure in /home/vagrant/.pip/pip.log
```

**Comment by SEJeff** [comment]
Missing test coverage sucks

**Pull request merged**